modules/mono/glue/GodotSharp/GodotSharp/Core/DelegateUtils.cs

```diff
@@ -461,7 +461,7 @@ private static bool TryDeserializeSingleDelegate(byte[] buffer, [MaybeNullWhen(f
 
                             if (fieldInfo != null)
                             {
-                                var variantValue = GD.BytesToVar(valueBuffer);
+                                var variantValue = GD.BytesToVarWithObjects(valueBuffer);
```

Comment:
So I looked at this again, it seems the `resource_path` property of `Resource` types (such as `CSharpScript`) is not serialized because it doesn't have the `PROPERTY_USAGE_STORAGE` flag.

So the path to the script is lost and we can't instantiate it later, the `script/source` fake property is stored so I assume this is how they make it work in GDScript, but C# doesn't support built-in scripts.

I don't think just adding the `PROPERTY_USAGE_STORAGE` flag to the `resource_path` property would make it work. Because then the problem is you'll likely run into the error `Another resource is loaded from path` or `An item with the same key has already been added (ScriptTypeBiMap)`.

I've also tested the `GD.Var2Bytes` and `GD.Bytes2Var` methods in 3.x and got the same error, so I guess this never really worked.

Reply:
That doesn't surprise me much. It's a kinda niche use case to begin with. Feel free to tell me if you think this PR still brings something to the table or not. I can remove the draft state if we want to merge this at some point. And/or we probably want to look in another direction regarding what to do with that kind of signal connections when reloading...

Reply:
I think the PR looks good (and fixes some parts of the old code that are definetely wrong) so I'm leaning towards merging it, but we'd need to wait for #78219 to be merged first since we depend on that to fix the `GD.VarToBytes{WithObjects}`/`GD.BytesToVar{WithObjects}` serialization with C# scripts. Since #78219 has been pushed to 4.3, I've pushed this PR to the 4.3 milestone as well.